client/src/app/menu/menu.component.ts

```diff
@@ -130,4 +131,18 @@ export class MenuComponent implements OnDestroy, OnInit {
         this.mobileQuery.removeListener(this._mobileQueryListener);
     }
 
+    isVisible(subMenu: SubMenu): Boolean {
+        for (const link of subMenu.links) {
+            if (link.permissions) {
+                for (const perm of link.permissions) {
+                    if (Object.keys(this.ngxPermissionsService.getPermissions()).indexOf(<string>perm) !== -1) {
+                        return true;
+                    }
+                }
+            } else {
```

Comment:
Maybe revert the condition to skip another branch (which grow the cyclic complexity) like:

```js
if (!link.permissions) return true;

for (...) {
	if (... !== -1) { // Maybe even use includes
		return true;	
	}
}
```